docs/user/vsphere/install_upi.md

```diff
@@ -279,72 +279,27 @@ The Ignition config created by the OpenShift Installer cannot be used directly b
   },
 }
 ```
-### Hostname
 
-The hostname of each control plane and worker machine must be resolvable from all nodes within the cluster.
+### Static hostname and IP Addresses via guestinfo
 
-Preferrably, the hostname will be set via DHCP. If you need to manually set a hostname, you can create a hostname file by adding an entry in the `.storage.files` list in an Ignition config.
+Preferably, the hostname and IP address for each machine will be set via DHCP; please see the vSphere documentation to enable that.  However, some organizations require static IP addressing.
 
-For example, the following Ignition config will create a hostname file that sets the hostname of a machine to `control-plane-0.
+As of OpenShift 4.6 for vSphere environments, the CoreOS "afterburn" component will lookup for the well-known key `guestinfo.afterburn.initrd.network-kargs` and use its value instead of the default.
+The property must be set on the VM before the first boot; changing the property after the instance is booted the first time will have no effect.
 
-```json
-{
-  "ignition": {
-    "version": "3.1.0"
-  },
-  "storage": {
-    "files": [
-      {
-        "path": "/etc/hostname",
-        "contents": {
-          "source": "data:text/plain;charset=utf-8,control-plane-0",
-        },
-        "mode": 420
-      }
-    ]
-  },
-}
-```
+The example below shows a `node1` VM booting with a static hostname and IPv4 routing on the `ens192` interface.
 
-### Static IP Addresses
+```
+VM_NAME="node1"
 
-Preferrably, the IP address for each machine will be set via DHCP. If you need to use a static IP address, you can be set one for a machine by creating an ifcfg file. You can create an ifcfg file by adding an entry in the `.storage.files` list in an Ignition config.
+IPCFG="ip=10.20.30.42::10.20.30.254:255.255.255.0:mynode01:ens192:off"
 
-For example, the following Ignition config will create an ifcfg file that sets the IP address of the ens192 device to 10.0.0.2.
+govc vm.change -vm "${VM_NAME}" -e "guestinfo.afterburn.initrd.network-kargs=${IPCFG}"
 
-```json
-{
-  "ignition": {
-    "version": "3.1.0"
-  },
-  "storage": {
-    "files": [
-      {
-        "path": "/etc/sysconfig/network-scripts/ifcfg-ens192",
-        "contents": {
-          "source": "data:text/plain;charset=utf-8;base64,VFlQRT1FdGhlcm5ldApCT09UUFJPVE89bm9uZQpOQU1FPWVuczE5MgpERVZJQ0U9ZW5zMTkyCk9OQk9PVD15ZXMKSVBBRERSPTEwLjAuMC4yClBSRUZJWD0yNApHQVRFV0FZPTEwLjAuMC4xCkRPTUFJTj1teWRvbWFpbi5jb20KRE5TMT04LjguOC44",
-        },
-        "mode": 420
-      }
-    ]
-  },
-}
+govc vm.power -on "${VM_NAME}"
```

Comment:
`govc` is a community tool (offered by vsphere). 
Who supports the use/distribution and maintenance of this tool?

Reply:
I think UPI installations may necessarily involve some tools not shipped by Red Hat.  The goal here is more to outline suggested approaches, not to *require* any specific tooling.  However we probably should have guidance on this.  I don't have expertise in vSphere enough to say, but I'm sure there are alternatives to using `govc` that we could discuss in a separate section.